pkg/aws/aws.go

```diff
@@ -142,6 +143,23 @@ func New(config *Config) (*AWS, error) {
 			}
 			collector := eks.New(config.Region, config.Profile, config.ScrapeInterval, pricingService, computeService, regions.Regions, regionClientMap)
 			collectors = append(collectors, collector)
+		case "EC2":
+			pricingService := pricing.NewFromConfig(ac)
+			computeService := ec2.NewFromConfig(ac)
```

Comment:
Question: since these need not be independent of the `eks` collector, should we consider creating `pricingService` and `computeService` once and just passing them to the constructor for either `eks` or `ec2` ?

I realize that you could define one or the other, and not necessarily both, but perhaps initializing them outside the `switch` statement might then allow them to be defined just once.

Reply:
I don't have a strong opinion. I was going back and forth on this and opted to create them twice. I opted for twice because there's no harm in them being created twice, but it feels a bit wrong to instantiate them if they're not going to be used.